terraform/modules/iam/templates/custom/s3_check_missing_ssl_policy.json.tpl

```diff
@@ -0,0 +1,29 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetBucketAcl",
+        "s3:GetBucketLocation",
+        "s3:GetBucketLogging",
+        "s3:GetBucketNotification",
+        "s3:GetBucketObjectLockConfiguration",
+        "s3:GetBucketPolicy",
+        "s3:GetBucketPolicyStatus",
+        "s3:GetBucketTagging",
+        "s3:GetBucketVersioning",
+        "s3:GetBucketWebsite",
+        "s3:GetEncryptionConfiguration",
+        "s3:GetLifecycleConfiguration",
+        "s3:GetReplicationConfiguration",
+        "s3:ListAllMyBuckets",
+        "s3:ListBucket",
+        "s3:ListBucketVersions",
+        "s3:PutBucketTagging",
+        "s3:PutBucketPolicy"
+      ],
+      "Resource": "*"
```

Comment:
Is it possible to use anything a bit more restrictive? Just thinking on the pen testing findings.

Reply:
I'm not sure we can. If we put in a list of our buckets then it will catch it if one of the existing buckets has the policy removed but won't catch any new ones. The only thing I can think is to have it like `arn:aws:s3:::tdr-*` and then if we get legitimate new buckets it should catch those. But then it won't catch anything called `test-xxx` I'm not sure. What do you think?

Reply:
Tricky, I think the wider permissions is worth it, given the benefit of catching these. So leave as is.